ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -478,3 +485,84 @@ func (scbuilder *SiteConfigBuilder) splitYamls(yamls []byte) ([][]byte, error) {
 
 	return resources, nil
 }
+
+func filterExtraManifests(dataMap map[string]interface{}, filter *Filter) (map[string]interface{}, error) {
+	// return if there's no filter initialized
+	if filter == nil {
+		return dataMap, nil
+	}
+
+	// check if there is an intersection in filenames
+	if filter.Include != nil && filter.Exclude != nil && len(filter.Exclude) > 0 && len(filter.Include) > 0 {
+		m := make(map[string]bool)
+		var fileExistsInBothLists []string
+
+		for _, includeFileName := range filter.Include {
+			m[includeFileName] = true
+		}
+
+		for _, excludeFileName := range filter.Exclude {
+			if _, ok := m[excludeFileName]; ok {
+				fileExistsInBothLists = append(fileExistsInBothLists, excludeFileName)
+			}
+		}
+
+		if len(fileExistsInBothLists) > 0 {
+			stringFiles := strings.Join(fileExistsInBothLists, ",")
+			errStr := fmt.Sprintf("Can not have same file names in both exclude and include lists. Files %s are present in both include and exclude lists", stringFiles)
+			return dataMap, errors.New(errStr)
+		}
+	}
+
+	inclusionDefaultInclude := "include"
+	inclusionDefaultExclude := "exclude"
+	// default value is include. treat use of `exclude` as an advanced feature
+	if filter.InclusionDefault == nil {
+		filter.InclusionDefault = &inclusionDefaultInclude
+	} else if strings.EqualFold(*filter.InclusionDefault, inclusionDefaultExclude) {
+		filter.InclusionDefault = &inclusionDefaultExclude
+	} else if strings.EqualFold(*filter.InclusionDefault, inclusionDefaultInclude) {
+		filter.InclusionDefault = &inclusionDefaultInclude
+	} else {
+		errStr := fmt.Sprintf("acceptable values for inclusionDefault are %s and %s. You have entered %s", inclusionDefaultInclude, inclusionDefaultExclude, *filter.InclusionDefault)
+		return dataMap, errors.New(errStr)
+	}
+
+	getDataMapFileNameInStrings := func(dataMap map[string]interface{}) string {
+		var files []string
+		for s := range dataMap {
+			files = append(files, s)
+		}
+		stringFiles := strings.Join(files, ",")
+		return stringFiles
+	}
+
+	// remove the files list
+	for _, fileToExclude := range filter.Exclude {
+		_, exists := dataMap[fileToExclude]
+		if exists {
+			delete(dataMap, fileToExclude)
+		} else {
+			errStr := fmt.Sprintf("Filename %s under exclude array is invalid. Valid files names are: %s", fileToExclude, getDataMapFileNameInStrings(dataMap))
+			return dataMap, errors.New(errStr)
+		}
+	}
+
+	// when `exclude` the final result will contain things in include list
+	if strings.EqualFold(*filter.InclusionDefault, inclusionDefaultExclude) {
+		temp := make(map[string]interface{})
+		for _, fileToInclude := range filter.Include {
+			value, exists := dataMap[fileToInclude]
+			if exists {
+				temp[fileToInclude] = value
+			} else {
+				errStr := fmt.Sprintf("Filename %s under include array is invalid. Valid files names are: %s",
+					fileToInclude, getDataMapFileNameInStrings(dataMap))
+				return dataMap, errors.New(errStr)
+			}
+		}
+		return temp, nil
```

Comment:
If you move this whole block up before the previous one, I think it might flow better:

1. if we're in "exclude" mode, do this block and return the copied  `temp` map.

2. If we bypass this short-circuit return, we know we're in "include" mode...  So just delete any entries in the exclude list from dataMap (line 541-549) and return the dataMap.

Reply:
done